sched/sched/sched_suspend.c

```diff
@@ -172,9 +176,13 @@ void nxsched_suspend(FAR struct tcb_s *tcb)
         {
           switch_needed = nxsched_remove_readytorun(tcb);
 
-          if (list_pendingtasks()->head)
+          if (!nxsched_islocked_tcb(rtcb) || switch_needed)
```

Comment:
why not check switch_needed first

Reply:
sure can, this form was just due to an earlier review comment. I'll check switch needed first!